GitHub Action tests fail

Since the merge of #1707 the [GitHub Action tests](https://github.com/ESMValGroup/ESMValTool/runs/2131703731?check_suite_focus=true) fail:


```
 =========================== short test summary info ============================
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg1] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg0] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg3] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg2] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg5] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg4] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg7] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg6] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg9] - ...
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg8] - ...
FAILED tests/integration/test_recipe_filler.py::test_not_adding_datasets[cfg0]
FAILED tests/integration/test_recipe_filler.py::test_adding_datasets[cfg10]
FAILED tests/integration/test_recipe_filler.py::test_bad_var - FileNotFoundEr...
FAILED tests/integration/test_recipe_filler.py::test_not_adding_datasets[cfg1]
ERROR esmvaltool/cmorizers/obs/cmorize_obs_cds_uerra.py - ImportError: cannot...
ERROR esmvaltool/cmorizers/obs/cmorize_obs_cds_uerra.py - ImportError: cannot...
= 14 failed, 1437 passed, 1 skipped, 233 warnings, 2 errors in 369.66s (0:06:09) =
Error: Process completed with exit code 1.
```

However, the corresponding [Circle CI tests](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValTool/4558/workflows/6b11b375-056b-43ee-b69e-4d4d21eb70a1) pass...

--- a/tests/integration/test_recipe_filler.py
+++ b/tests/integration/test_recipe_filler.py
@@ -1,13 +1,15 @@
 """Tests for _data_finder.py."""
 import contextlib
 import os
-import sys
 import shutil
+import sys
 import tempfile
 
 import pytest
 import yaml
+from packaging import version
 
+import esmvalcore
 import esmvaltool.utils.recipe_filler as recipe_filler
 from esmvalcore.cmor.table import read_cmor_tables
 from esmvaltool.utils.recipe_filler import run
@@ -119,7 +121,10 @@ def setup_files(tmp_path, root, cfg):
     return user_config_file, recipe, output_recipe
 
 
-@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
+@pytest.mark.skipif(
+    version.parse(esmvalcore.__version__) < version.parse("2.3.0"),
+    reason="reactivate for esmvalcore=2.3.0"
+)
 @pytest.mark.parametrize('cfg', CONFIG['has_additional_datasets'])
 def test_adding_datasets(tmp_path, root, cfg):
     """Test retrieving additional datasets."""
@@ -145,7 +150,10 @@ def test_adding_datasets(tmp_path, root, cfg):
         assert "additional_datasets" in var
 
 
-@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
+@pytest.mark.skipif(
+    version.parse(esmvalcore.__version__) < version.parse("2.3.0"),
+    reason="reactivate for esmvalcore=2.3.0"
+)
 @pytest.mark.parametrize('cfg', CONFIG['no_additional_datasets'])
 def test_not_adding_datasets(tmp_path, root, cfg):
     """Test retrieving no additional datasets."""
@@ -171,7 +179,10 @@ def test_not_adding_datasets(tmp_path, root, cfg):
         assert "additional_datasets" not in var
 
 
-@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
+@pytest.mark.skipif(
+    version.parse(esmvalcore.__version__) < version.parse("2.3.0"),
+    reason="reactivate for esmvalcore=2.3.0"
+)
 def test_bad_var(tmp_path, root):
     """Test a bad variable in the works."""
     cfg = CONFIG['bad_variable'][0]

--- a/tests/integration/test_recipe_filler.py
+++ b/tests/integration/test_recipe_filler.py
@@ -119,6 +119,7 @@ def setup_files(tmp_path, root, cfg):
     return user_config_file, recipe, output_recipe
 
 
+@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
 @pytest.mark.parametrize('cfg', CONFIG['has_additional_datasets'])
 def test_adding_datasets(tmp_path, root, cfg):
     """Test retrieving additional datasets."""
@@ -144,6 +145,7 @@ def test_adding_datasets(tmp_path, root, cfg):
         assert "additional_datasets" in var
 
 
+@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
 @pytest.mark.parametrize('cfg', CONFIG['no_additional_datasets'])
 def test_not_adding_datasets(tmp_path, root, cfg):
     """Test retrieving no additional datasets."""
@@ -169,6 +171,7 @@ def test_not_adding_datasets(tmp_path, root, cfg):
         assert "additional_datasets" not in var
 
 
+@pytest.mark.skip(reason="reactivate for esmvalcore=2.3.0")
 def test_bad_var(tmp_path, root):
     """Test a bad variable in the works."""
     cfg = CONFIG['bad_variable'][0]

